From 0315cef04a5a8a953072691faa48af9acb6009bd Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 10 Aug 2021 13:39:58 +0200
Subject: [PATCH] include/flatbuffers/base.h: fix build on musl

Build of applications using flatbuffers such as snort3 are broken on
musl since version 1.11.0 because strtoll_l and strtoull_l are not
available on musl.

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 include/flatbuffers/base.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/flatbuffers/base.h b/include/flatbuffers/base.h
index de7898dc..101c7598 100644
--- a/include/flatbuffers/base.h
+++ b/include/flatbuffers/base.h
@@ -270,7 +270,8 @@ namespace flatbuffers {
   // strtoull_l}.
   #if (defined(_MSC_VER) && _MSC_VER >= 1800) || \
       (defined(__ANDROID_API__) && __ANDROID_API__>= 21) || \
-      (defined(_XOPEN_VERSION) && (_XOPEN_VERSION >= 700)) && \
+      (defined(_XOPEN_VERSION) && (_XOPEN_VERSION >= 700) && \
+        defined(__GLIBC__) && !defined(__UCLIBC__)) && \
         (!defined(__Fuchsia__) && !defined(__ANDROID_API__))
     #define FLATBUFFERS_LOCALE_INDEPENDENT 1
   #else
-- 
2.30.2

